mxcubeweb/core/server/resource_handler.py

```diff
@@ -0,0 +1,500 @@
+import contextlib
+import logging
+import re
+from collections.abc import Callable
+from functools import reduce
+from typing import ClassVar
+
+from flask import (
+    Blueprint,
+    jsonify,
+    request,
+)
+from pydantic.v1 import (
+    BaseModel,
+    ValidationError,
+)
+
+from mxcubeweb.core.server.openapidoc import OpenAPISpec
+
+
+def valid_object_id(object_id: str) -> bool:
+    """
+    Validates that the object_id contains only A-Z, a-z, and '.' , '_'
+    Args:
+        input_string (str): The string to validate.
+
+    Returns:
+        bool: True if the string is valid, False otherwise.
+    """
+    return bool(re.fullmatch(r"[A-Za-z._]+", object_id))
```

Comment:
Not sure where the `input_string` comes from.

There has to be an empty line.

```suggestion
def valid_object_id(object_id: str) -> bool:
    """Validate that ``object_id`` contains only ``A-Z``, ``a-z``, and ``.`` , ``_``.
    
    Args:
        object_id (str): The string to validate.

    Returns:
        bool: ``True`` if the string is valid, ``False`` otherwise.
    """
    return bool(re.fullmatch(r"[A-Za-z._]+", object_id))
```

Reply:
The `input_string` comes from me writing doc string a bit carelessly, thanks :)